src/common/dateHelpers.js

```diff
@@ -17,18 +18,25 @@ const getCurrentDate = () =>
  * @param {Date} params.date Start date, typically the current date.
  * @param {number} params.subtract Number of granularity type to subtract from the current date.
  * @param {string} params.measurement Granularity type.
+ * @param {string} params.endOfMeasurement Granularity type.
  * @returns {{endDate: Date, startDate: Date}}
  */
-const setRangedDateTime = ({ date, subtract, measurement }) => ({
+const setRangedDateTime = ({ date, subtract, measurement, endOfMeasurement = 'days' }) => ({
   startDate: moment.utc(date).startOf(measurement).subtract(subtract, measurement).toDate(),
-  endDate: moment.utc(date).startOf(measurement).endOf('days').toDate()
+  endDate: moment.utc(date).startOf(measurement).endOf(endOfMeasurement).toDate()
 });
 
 const currentDateTime = setRangedDateTime({ date: getCurrentDate(), subtract: 1, measurement: 'days' });
 const defaultDateTime = setRangedDateTime({ date: getCurrentDate(), subtract: 30, measurement: 'days' });
 const weeklyDateTime = setRangedDateTime({ date: getCurrentDate(), subtract: 12, measurement: 'weeks' });
 const monthlyDateTime = setRangedDateTime({ date: getCurrentDate(), subtract: 12, measurement: 'months' });
 const quarterlyDateTime = setRangedDateTime({ date: getCurrentDate(), subtract: 36, measurement: 'months' });
+const rangedYearDateTime = setRangedDateTime({
+  date: getCurrentDate(),
+  subtract: 11,
+  measurement: 'months',
+  endOfMeasurement: 'months'
```

Comment:
required: confirm this shouldn't be the last day of the month instead of the first day of the next month

Reply:
Updated to be the end of the month instead of the first of the following month